src/policy.rs

```diff
@@ -1,31 +1,15 @@
-use crate::attribute::Attribute;
-use crate::configuration::{Action, DataItem, DataType, PatternExpression};
-use crate::envoy::{RateLimitDescriptor, RateLimitDescriptor_Entry};
-use crate::filter::http_context::Filter;
-use log::debug;
-use proxy_wasm::traits::Context;
+use crate::configuration::{Action, PatternExpression};
 use serde::Deserialize;
 
-#[derive(Deserialize, Debug, Clone)]
-#[serde(rename_all = "camelCase")]
-pub struct Condition {
-    pub all_of: Vec<PatternExpression>,
-}
-
 #[derive(Deserialize, Debug, Clone)]
 pub struct Rule {
-    //
-    #[serde(default)]
-    pub conditions: Vec<Condition>,
-    //
-    pub data: Vec<DataItem>,
+    pub conditions: Vec<PatternExpression>,
```

Comment:
@eguzki preferred retaining the name `allOf` over `conditions` here to be explicit that all conditions must apply. I myself am of the opinion that `allOf` suggests there's an `anyOf` so prefer conditions, so would like other thoughts

Reply:
I think it will eventually (very soon) be called `predicates`, right @alexsnaps ?

Reply:
I think whether these are `Predicate`s or `PatternExpression`s here is orthogonal to @adam-cattermole 's point. I _think_ he wants to keep the field names `allOf`, which indeed eventually will be something else than `PatternExpression`. It would eventually become a cel expression representing a [`Predicate`](https://en.wikipedia.org/wiki/Propositional_function) indeed, tho, as discussed, how that would look like is probably dependent on what work the Kuadrant Operator does with the initial unparsed expression... if it is parsed entirely, we could inject the AST straight into here (more so if there is an easy way to (de)serialize it to yaml, from Go to Rust) or just pass the `String` around...